app/src/main/java/org/dslul/openboard/inputmethod/latin/utils/ActivityThemeUtils.java

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-3.0-only
+package org.dslul.openboard.inputmethod.latin.utils;
+
+import android.os.Build;
+import android.view.View;
+import android.view.WindowInsetsController;
+
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.app.AppCompatActivity;
+
+public class ActivityThemeUtils {
+
+    public static void setActivityTheme(final AppCompatActivity activity) {
+        final boolean isNight = ResourceUtils.isNight(activity.getResources());
+        final ActionBar actionBar = activity.getSupportActionBar();
+        if (actionBar == null) {
+            return;
+        }
```

Comment:
Why is the action bar necessary for the code below? It doesn't seem directly related.

Reply:
This is linked to commit eadc83d.
Without this code, it is impossible to launch `SpellCheckerSettingsActivity`.

Reply:
I deleted it with commit 7f49aba. Is it ok now?

Reply:
It was also ok before. If it's needed for something, you can just add a comment. This is quite important as just from the `setActivityTheme` function the reason for this code is completely unclear.

The current version now contains `setActivityTheme` and `setFragmentTheme`, which contain exactly the same code.
I guess you could merge them and just use an `Activity` as parameter.

Reply:
> If it's needed for something, you can just add a comment.

Really sorry, I completely forgot to mention it when I wanted to write it.

> The current version now contains setActivityTheme and setFragmentTheme, which contain exactly the same code.

😮😮The other day it didn't work and now it does.
Weird but everything is ok 👍
Commit 8c91dc6